pkg/apply/applier.go

```diff
@@ -159,6 +160,21 @@ func (a *Applier) Run(ctx context.Context, invInfo inventory.InventoryInfo, obje
 		}
 		klog.V(4).Infof("calculated %d apply objs; %d prune objs", len(applyObjs), len(pruneObjs))
 
+		// Add all objects to a dependency graph
```

Comment:
This chunk makes me cringe. 

I wanted to just use `objects` but that doesn't include the objects in the inventory that need pruning. So I had to merge `applyObjs` and `pruneObjs` without modifying either set.

Then the AddDependencies functions can only be executed once, otherwise it can't resolve all the dependencies. In order to make AddDependencies executable multiple times it would need to be pulled out to its own class with a Build or Finalize function to validate if there are any external dependencies...